Core/Scripts: Teron Gorefiend encounter rewrite

Keader opened the pull request:
Rewrite Teron Gorefiend Encounter - Black Temple

**Changes proposed:**

-  All fights rewritten
-  Replaced summons hacks with proper spells
-  Fixed Ghost (Vengeful Spirit)
- Fixed immunities for Shadowy Constructs, script rewritten
- Doom Blossom script fixed/rewritten
- Added missing spells
- Added spell scripts: Shadow of Death, Shadow of Death Remove, Spiritual Vengeance
- Fixed creature texts
- Fixed spell timers
- Added DespawnAtEvade method
- Replaced old hack for event intro with proper area trigger

**Target branch(es):** 3.3.5/master

**Issues addressed:** (no open ticket)

**Tests performed:** Everything built and tested in game.

**Many thanks to:** sirikfoll, ariel-, aokromes, ccrs, riztazz, joschiwald, funjoker for all help 👍

Keader committed:
Core/Scripts: Rewrite Teron Gorefiend Encounter

Keader committed:
Added Player check in KilledUnit

Keader committed:
Making Sirikfoll happy (Identation)

Keader committed:
No Copyright for you anymore  :)

offl commented on sql/updates/world/3.3.5/9999_99_52_00_world_335.sql:
```diff
@@ -0,0 +1,35 @@
+DELETE FROM `creature_text` where `entry`=22871;
+INSERT INTO `creature_text` (`entry`, `groupid`, `id`, `text`, `type`, `language`, `probability`, `emote`, `duration`, `sound`, `BroadcastTextId`, `TextRange`, `comment`) VALUES
+(22871, 0, 0, 'I was the first you know. For me the wheel of death has spun many times. So much time has passed... I have a lot of catching up to do.' , 14, 0, 100, 0, 0, 11512, 21098, 0, 'Teron SAY_INTRO'),
+(22871, 1, 0, 'Vengeance is mine!', 14, 0, 100, 0, 0, 11513, 28969, 0, 'Teron SAY_AGGRO'),
```
The correct BroadcastTextId is 21097.

Keader commented:
@joschiwald i trying add a missing emote, but instance->GetCreature() returns null
Help please https://gist.github.com/Keader/bea7f439aec9ac257d4cb761987aaf26

Keader committed:
typo

Commit:
Update 9999_99_52_00_world.sql

Teron Gorefiend creature_text comment improvements,
keeping comment content on par with TC code style.

Keader committed:
Avoid intro start when teron is dead or in DespawnAtEvade

funjoker commented on sql/updates/world/3.3.5/9999_99_52_00_world.sql:
```diff
@@ -0,0 +1,35 @@
+
+-- Area Trigger
+DELETE FROM `areatrigger_scripts` WHERE `entry`=4665;
+INSERT INTO `world`.`areatrigger_scripts` (`entry`, `ScriptName`) VALUES
```
`world`. <<<<<<<<

funjoker reviewed (changes requested):
Remove the `world.` DB reference (to match TC codestyle).

Keader committed:
typo

Keader commented:
Emote issue handled here: https://github.com/TrinityCore/TrinityCore/pull/18293

Keader committed:
Transform cast of some auras in creature_template_addon

Aokromes committed:
Rename 9999_99_52_00_world.sql to 2016_11_26_04_world.sql
